resource/subfunctions/resource_class.md

```diff
@@ -0,0 +1,323 @@
+
+#### 📤 Returns
+* `Union[dict, bytes]`<br>
+  Returns a **Security Result dictionary** or **Security Request XML bytes** if the `ResourceAdmin.generate_requests_only` class attribute is set to `True`.
+
+#### ❌ Raises
+* `SecurityRequestError`<br>
+  Raises `SecurityRequestError` when the **Return Code** of a **Security Result** returned by IRRSMO00 is **NOT** equal to `0`.
+* `AlterOperationError`<br>
+  Raises `AlterOperationError` when the **z/OS userid** supplied cannot be altered because it does **NOT** exist.
+* `SegmentTraitError`<br>
+  Raises `SegmentTraitError` when the dictionary of **traits/attributes** provided contains one or more **unknown** traits.
+
+#### 💻 Example
+
+###### Python REPL
+```python
+from pyracf import ResourceAdmin
+resource_admin = ResourceAdmin()
+class_traits = {
+    "cdtinfo:valid_first_characters": ["ALPHA", "NUMERIC"],
+    "cdtinfo:valid_other_characters": ["ALPHA"],
+    "cdtinfo:profiles_allowed": "NO",
+    "cdtinfo:default_racroute_return_code": 4,
+    "cdtinfo:default_universal_access": "READ",
+}
+resource_admin.alter_resource_class("SHELCITY",traits=class_traits)
+{'securityResult': {'resource': {'name': 'SHELCITY', 'class': 'CDT', 'operation': 'set', 'requestId': 'ResourceRequest', 'info': ['Definition exists. Add command skipped due  to precheck option'], 'commands': [{'safReturnCode': 0, 'returnCode': 0, 'reasonCode': 0, 'image': 'RALTER  CDT             (SHELCITY)  CDTINFO  (FIRST       (ALPHA NUMERIC) OTHER       (ALPHA) PROFILESALLO(NO) DEFAULTRC   (4) DEFAULTUACC (READ))', 'messages': ['IRR52199I Warning: Class name SHELCITY does not contain a national character nor a number.', 'ICH11009I RACLISTED PROFILES FOR CDT WILL NOT REFLECT THE UPDATE(S) UNTIL A SETROPTS REFRESH IS ISSUED.']}]}, 'returnCode': 0, 'reasonCode': 0}}
+```
+
+###### Security Result Dictionary as JSON
+```json
+{
+  "securityResult":{
+    "resource":{
+      "name":"SHELCITY",
+      "class":"CDT",
+      "operation":"set",
+      "requestId":"ResourceRequest",
+      "info":[
+        "Definition exists. Add command skipped due  to precheck option"
+      ],
+      "commands":[
+        {
+          "safReturnCode":0,
+          "returnCode":0,
+          "reasonCode":0,
+          "image":"RALTER  CDT             (SHELCITY)  CDTINFO  (FIRST       (ALPHA NUMERIC) OTHER       (ALPHA) PROFILESALLO(NO) DEFAULTRC   (4) DEFAULTUACC (READ))",
+          "messages":[
+            "IRR52199I Warning: Class name SHELCITY does not contain a national character nor a number.",
+            "ICH11009I RACLISTED PROFILES FOR CDT WILL NOT REFLECT THE UPDATE(S) UNTIL A SETROPTS REFRESH IS ISSUED."
+          ]
+        }
+      ]
+    },
+    "returnCode":0,
+    "reasonCode":0
+  }
+}
+```
+
+## `ResourceAdmin.extract_resource_class()`
+
+```python
+  def extract_resource_class( self, class_name: str ) -> Union[dict, bytes]:
+```

```

Comment:
```python
def extract_resource_class(self, class_name: str) -> Union[dict, bytes]:
```

Reply:
Leading spaces and unnecessary trailing spaces and indents removed across all function definitions.

Reply:
change `extract_resource_class(self, class_name: str )` to `extract_resource_class(self, class_name: str)`

Reply:
Sorry, somehow class also got missed. now fixed in extract and delete.